### ISSUE TITLE
App archive fails validation due to invalid plist file

### ISSUE THREAD

**Issue by cameroncooke** [firebase/firebase-ios-sdk#12441]
### Description

When building our app for Archive using Xcode 15.3 the app fails the validation stage on App Store Connect due to problems with the framework's `Info.plist` file. Our app is targeting iOS and has a deployment target of iOS 16.

It's worth noting that the same is happening for GoogleAppMeasurement (https://github.com/google/GoogleAppMeasurement/issues/62) library and another 3rd party library in our project.

### Reproducing the issue

Build an app using Xcode 15.3 RC that uses the Google Firebase SPM package

### Firebase SDK Version

10.18.0

### Xcode Version

15.3 RC

### Installation Method

Swift Package Manager

### Firebase Product(s)

All

### Targeted Platforms

iOS

### Relevant Log Output

```shell
Error
Missing Info.plist value. A value for the key 'MinimumOSVersion' in bundle MyApp.app/Frameworks/FirebaseAnalytics.framework is required.

Error
Invalid Bundle. The bundle MyApp.app/Frameworks/FirebaseAnalytics.framework does not support the minimum OS Version specified in the Info.plist.

Error
The bundle 'Payload/MyApp.app/Frameworks/FirebaseAnalytics.framework' is missing plist key. The Info.plist file is missing the required key: CFBundleShortVersionString. Please find more information about CFBundleShortVersionString at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring

Error
Invalid MinimumOSVersion. Apps that only support 64-bit devices must specify a deployment target of 8.0 or later. MinimumOSVersion in 'MyApp.app/Frameworks/FirebaseAnalytics.framework' is ''.
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by cameroncooke** [comment]
When will this release be made available?

**Comment by ncooke3** [comment]
This was fixed by #12441. And subsequent issues have been addressed in #12439. These fixes will be released in the next Firebase release (week of March 5).

**Comment by paulb777** [comment]
In the meantime, the workaround is to continue to use Xcode 15.2 or earlier.

**Comment by dannycortesv** [comment]
I'm using Xcode 14.2 and also getting this issue...

**Comment by nsarno-lucid** [comment]
I'm using Xcode 15.2 and getting this issue on all App Store Connect archives. Is there another fix for this? Is there an older version of Firebase we should use in the meantime?

**Comment by paulb777** [comment]
If you're having a validation problem with an Xcode other than 15.3, please open a separate issue with the template filled out describing the problem and installation method.

**Comment by nsarno-lucid** [comment]
Apologies, you are correct above in that it works on Xcode 15.2. 

In case anyone else runs into this... I did not realize Xcode Cloud archives do not use the Xcode version that is on your computer. You can set the Xcode version for Xcode Cloud builds within App Store Connect -> Xcode Cloud -> Manage Workflows -> Workflow -> Xcode version. The default value is "Latest Release" which is currently Xcode 15.3 RC.

**Comment by andrewheard** [comment]
@cameroncooke This should now be fixed in the [10.22.0 release](https://github.com/firebase/firebase-ios-sdk/releases/tag/10.22.0). Please re-open if you're still running into app validation issues.

**Comment by trevordevelops** [comment]
> @cameroncooke This should now be fixed in the [10.22.0 release](https://github.com/firebase/firebase-ios-sdk/releases/tag/10.22.0). Please re-open if you're still running into app validation issues.

it is not fixed for me with 15.3, I am getting numerous 'asset validation failed' errors on distribution via archive.

<img width="1562" alt="Screenshot 2024-03-05 at 6 40 17 PM" src="https://github.com/firebase/firebase-ios-sdk/assets/85640211/f85d2399-de34-4fb9-a6da-aa043a07fe6d">

**Comment by cameroncooke** [comment]
> > @cameroncooke This should now be fixed in the [10.22.0 release](https://github.com/firebase/firebase-ios-sdk/releases/tag/10.22.0). Please re-open if you're still running into app validation issues.
> 
> 
> 
> it is not fixed for me with 15.3, I am getting numerous 'asset validation failed' errors on distribution via archive.
> 
> 
> 
> <img width="1562" alt="Screenshot 2024-03-05 at 6 40 17 PM" src="https://github.com/firebase/firebase-ios-sdk/assets/85640211/f85d2399-de34-4fb9-a6da-aa043a07fe6d">
> 
> 

That's a different library. Please contact the library vendor for a fix. The Firebase devs have fixed the issue for this library.

**Comment by lanserxt** [comment]
+1

**Comment by johnarnaou** [comment]
+1

**Comment by 404wasfound** [comment]
`Firebase` package itself does not seem to cause `Invalid MinimumOSVersion` error, but dependencies it brings with it (seems like these are `abseil` and `gRPC`) do.

**Comment by cameroncooke** [comment]
> `Firebase` package itself does not seem to cause `Invalid MinimumOSVersion` error, but dependencies it brings with it (seems like these are `abseil` and `gRPC`) do. 

Correct. Though you could argue that a package vendor is responsible for all the decencies it brings in. Apple even sees it this way when it comes to the new privacy manifests.

**Comment by johnauger** [comment]
+1 - Still experiencing validation failures on Xcode 15.3 after using 10.22.0...

**Comment by andrewheard** [comment]
@ncooke3 is working on a fix for the errors in the dependencies in https://github.com/google/abseil-cpp-binary/pull/6 and https://github.com/google/grpc-binary/pull/10.

**Comment by ncooke3** [comment]
> @ncooke3 is working on a fix for the errors in the dependencies in [google/abseil-cpp-binary#6](https://github.com/google/abseil-cpp-binary/pull/6) and [google/grpc-binary#10](https://github.com/google/grpc-binary/pull/10).

The binary SPM distributions for Abseil, gRPC, and BoringSSL have now been updated and released. The latest version of Firebase should pick them up if you update your SPM workspace. Each of the three SDKs also now has a privacy manifest.

**Comment by alexanderkhitev** [comment]
> > @ncooke3 is working on a fix for the errors in the dependencies in [google/abseil-cpp-binary#6](https://github.com/google/abseil-cpp-binary/pull/6) and [google/grpc-binary#10](https://github.com/google/grpc-binary/pull/10).
> 
> The binary SPM distributions for Abseil, gRPC, and BoringSSL have now been updated and released. The latest version of Firebase should pick them up if you update your SPM workspace. Each of the three SDKs also now has a privacy manifest.

waiting for an update, thanks!

**Comment by ncooke3** [comment]
> > > @ncooke3 is working on a fix for the errors in the dependencies in [google/abseil-cpp-binary#6](https://github.com/google/abseil-cpp-binary/pull/6) and [google/grpc-binary#10](https://github.com/google/grpc-binary/pull/10).
> > 
> > 
> > The binary SPM distributions for Abseil, gRPC, and BoringSSL have now been updated and released. The latest version of Firebase should pick them up if you update your SPM workspace. Each of the three SDKs also now has a privacy manifest.
> 
> waiting for an update, thanks!

@alexanderkhitev, the binary dependencies have been released. If you're using the binary distribution of Firestore (on by default), then you can pull them in by updating your SPM workspace.

**Comment by johnauger** [comment]
Thanks for this. What are the new version numbers for these? I'm seeing the following version numbers in the Xcode cloud build log:

Firebase: https://github.com/firebase/firebase-ios-sdk @ 10.22.0
GoogleAppMeasurement: https://github.com/google/GoogleAppMeasurement.git @ 10.22.0
abseil v1.2022062300.1 (abseil: https://github.com/google/abseil-cpp-binary.git @ 1.2022062300.1)
gRPC: https://github.com/google/grpc-binary.git @ 1.49.2


...and still getting the same error as above when building and validating with Xcode 15.3. 

Any thoughts? Thanks

**Comment by ncooke3** [comment]
> abseil v1.2022062300.1 (abseil: https://github.com/google/abseil-cpp-binary.git @ 1.2022062300.1)
> gRPC: https://github.com/google/grpc-binary.git @ 1.49.2

^ These should be the fixed versions. You shouldn't be seeing validation errors related to these frameworks anymore. Please let me know if you're seeing differently. 

> GoogleAppMeasurement: https://github.com/google/GoogleAppMeasurement.git @ 10.22.0

^ This one is still being worked on.

**Comment by helloniklas** [comment]
Yeah, I'm now getting this error.

> The bundle 'Payload/Motionscape-Alpha.app/Frameworks/GoogleAppMeasurementOnDeviceConversion.framework' is missing plist key. The Info.plist file is missing the required key: CFBundleShortVersionString. Please find more information about CFBundleShortVersionString at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring (ID: 29ccc9f2-b3a5-4866-b74e-6b8ea2f88ebf)

**Comment by ncooke3** [comment]
A fix for this should be coming out today. Binary artifacts are currently being rebuilt. Until then, a workaround should be to fall back to Xcode 15.2.

**Comment by andrewheard** [comment]
Unfortunately we need to delay the patch release until tomorrow. Please continue to use Xcode 15.2 in the meantime. Sorry for the delay in getting this out.

**Comment by dimitrihamelin** [comment]
No problem, you are doing some great job and this is the most important.

**Comment by ldlefebvre** [comment]
Getting the same issue with Payload GoogleAppMeasurementOnDeviceConversion.framework with key CFBundleShortVersionString. Looking forward to a fix :)

**Comment by andrewheard** [comment]
Sorry again for the delay. The fixes are now released:

- SPM users, please upgrade to version `10.22.1`.
- CocoaPods users, please update your pods to get `FirebaseAnalyticsOnDeviceConversion` `10.22.1`.

**Comment by kaisdewani** [comment]
I can confirm the new 10.22.1 has resolved the Payload GoogleAppMeasurementOnDeviceConversion.framework with key CFBundleShortVersionString missing error for me. 

I am using xcode 15.3

**Comment by johnauger** [comment]
Thanks for getting this sorted so promptly. Much appreciated. 👍🏻

**Comment by andrewheard** [comment]
> I can confirm the new 10.22.1 has resolved the Payload GoogleAppMeasurementOnDeviceConversion.framework with key CFBundleShortVersionString missing error for me.

Thanks for your patience and for confirming!

**Comment by lenli** [comment]
I've bumped my Firebase version to v10.22.1, but I'm still getting build errors on Xcode Cloud using the latest release of Xcode / MacOS:

<img width="1081" alt="Screenshot 2024-03-07 at 1 37 13 PM" src="https://github.com/firebase/firebase-ios-sdk/assets/799767/548bf222-b71b-44b0-964a-7e2f846ce30b">

The issue is with gRPC which was supposed to fixed in v10.22.0.
> abseil v1.2022062300.1 (abseil: https://github.com/google/abseil-cpp-binary.git @ 1.2022062300.1)
> gRPC: https://github.com/google/grpc-binary.git @ 1.49.2
<img width="329" alt="Screenshot 2024-03-07 at 1 35 53 PM" src="https://github.com/firebase/firebase-ios-sdk/assets/799767/bf8a4f34-cff3-4639-badd-a8b11cb2189a">

Is anyone else seeing this?

**Comment by dimitrihamelin** [comment]
Thank you so much for the update.

**Comment by ncooke3** [comment]
> TLDR: This fixes the GoogleAppMeasurement issue, but seems to have lost the fixes for gRPC and abseil.
Is anyone else seeing this?

Hi @lenli, I'm not sure why Xcode Cloud would be handling this differently. Perhaps there is interference from a checked-in `Package.resolved`?

**Comment by lenli** [comment]
> Hi @lenli, I'm not sure why Xcode Cloud would be handling this differently. Perhaps there is interference from a checked-in `Package.resolved`?

I think I figured out what happened:
If I open Xcode 15.3, I get the dependency graph that I posted above.
If I open the same project in Xcode 15.2, I get an updated `Package.resolved` with the correct dependencies.

**Comment by johnauger** [comment]
I just ran an Xcode Cloud build using 15.3, all good here.

Ensure packages are updated and the resulting Package.resolved is pushed to repo before starting build...

Thanks for a quick resolution. 👍🏻

**Comment by lenli** [comment]
Yep, it worked. 👍 
Just FYI, I did reset cache and upgrade to the latest dependencies, but for some reason Xcode 15.3 did not give me the latest.

**Comment by frantischek** [comment]
I updated, but still get this error: Couldn't stat /Users/*****/Library/Developer/CoreSimulator/Devices/****-****42B0-9841-*****/data/Library/Caches/com.apple.mobile.installd.staging/temp.aWbqKV/extracted/****/Frameworks/GoogleMobileAds.framework/Info.plist: No such file or directory 
<img width="281" alt="grafik" src="https://github.com/firebase/firebase-ios-sdk/assets/6368507/b60c0205-944a-4a46-b4f2-3e0aee55a8df">

**Comment by ncooke3** [comment]
@frantischek, I would recommend deleting derived data if you haven't already. Otherwise, please file an issue on the Ads repo (https://github.com/googleads/swift-package-manager-google-mobile-ads/issues).

**Comment by ncooke3** [comment]
To wrap up this issue's discussion, 
- If you are using SwiftPM  → Update to `10.22.1`.
- If you are using CocoaPods  → Run `pod update`.
- If you are using xcframeworks from the 10.22.0 zip distribution, choose "do not embed" when dragging in the  `GoogleAppMeasurementOnDeviceConversion` xcframework. 

If you run into any further issues, please open a new issue. Thanks!
